### DISCUSSION TITLE
Viewing images in ppixiv doesn't add them to history.php but does influence Recommended Works

### DISCUSSION THREAD

**Discussion by yaomtc** [ppixiv/ppixiv#90] [General]
I've noticed that when I start browsing an artist's work via ppixiv, those works don't appear in history.php, but it _does_ seem to influence the images shown in /discovery ("Recommended Works"), seemingly causing more of that artist's works to appear there next time. I am logged in, they have my IP address, and I guess their servers record that my IP accessed that artist's works? Maybe their recommendation system pulls from that internal data too, and not just bookmarks and history.php... Thoughts?

**Comment by ppixiv** [comment]
I haven't noticed viewing images influencing recommendations directly, but the API exposes some obscure info about how they generate suggestions:

- search, gender_ranking: straightforward
- illust_for_user_mf_vh_bookmark, illust_for_user_mf_vh_bookmark_recent: These two seem like "posts by users you've bookmarked images from before".  A couple artists I've bookmarked a lot of images from appear constantly in recommendations.
- bookmark_fresh_cos_tag: I think this is "recent images with tags similar to an image you've bookmarked".  It only specifies a single image, the ones below might be more sophisticated to find patterns across multiple images.
- clustering_bqalgc: "LGC" might be Local Graph Clustering, eg. "people who bookmarked that image also bookmarked this one".  This one is tied to a specific image.
- illust_for_user_tda000: Google says TDA might be "Topological Data Analysis" but I don't know anything further.  Seems like another heuristic looking for patterns in people with similar interests.  This one doesn't specify an image.  I don't know what the other inputs to these two might be.

**Comment by yaomtc** [comment]
Thanks for the insight!
